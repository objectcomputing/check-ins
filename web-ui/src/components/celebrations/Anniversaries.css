@import url('https://fonts.googleapis.com/css?family=Raleway:200');

body {
  font-family: 'Raleway', sans-serif;
  background-color: #F0E2DB;
  font-size: 20px;
  transition: 1s;
}

body.open{
  animation: bg-change 1s 1;
  background-color: #72c8d5;
  transition: 1s;
}
@keyframes bg-change{
  0%{
      background-color: #F0E2DB;
  }
  100%{
      background-color: #72c8d5;
  }
}
.anniversaries {
  display: flex;
  justify-content: center;
}
.gift {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 100px;
  height: 100px;
  transform-origin: 0% 0%;
  transform: translate(-50%, -50%);
  transition: 1s;
/*   animation: dukdik 0.5s infinite; */
}
.dukdik{
  transform-origin: 0% 0%;
  animation: dukdik 0.5s infinite;
}
@keyframes dukdik {
  0% {
    transform: rotate(0deg) translate(-50%, -50%);
  }
  25%{
    transform: rotate(1deg) translate(-50%, -50%);
  }
  50%{
    transform: rotate(0deg) translate(-50%, -50%);
  }
  70%{
    transform: rotate(-1deg) translate(-50%, -50%);
  }
  100%{
    transform: rotate(0deg) translate(-50%, -50%);
  }
}
.gift-top {
  position: relative;
  width: 110%;
  height: 25%;
  background: #2559a7;
  border-radius: 5px 5px 0 0;
  transform: translate3d(-5%, 0, 0);
  background-image: linear-gradient(to right, #2559a7 0%, #2559a7 44%, #feb672 44%, #feb672 56%, #2559a7 56%, #2559a7 100%);
  transition: 1s;
}
.gift-top.boxOpen{
  animation: box-open 1s;
  opacity:0;
}
@keyframes box-open{
  0%{
    top:0px;
    opacity:1;
  }
  100%{
    top:-50px;
    opacity:0;
  }
}
.gift-top::before,
.gift-top::after {
    content: "";
    width: 15px;
    height: 15px;
    border: 10px solid #feb672;
    border-radius: 50% 50% 0 50%;
    position: absolute;
    top: 0;
    left: 50%;
    transform: translate3d(-100%, -100%, 0) skew(10deg, 10deg);
}

.gift-top::after {
  border-radius: 50% 50% 50% 0;
  transform: translate3d(0, -100%, 0) skew(-10deg, -10deg);
}

.gift-text {
  position: absolute;
  top: 50%;
  left: 50%;
  margin: 0;
  color: #FFFFFF;
  white-space: nowrap;
  transform: translate3d(-50%, -50%, 0) scale(.5);
  transform-origin: center center;
  z-index: -1;
  opacity:1;
  transition: 5s;
}

.gift-box {
  width: 100%;
  height: 75%;
  position:relative;
  background-image: linear-gradient(to right, #2559a7 0%, #2559a7 45%, #feb672 45%, #feb672 55%, #2559a7 55%, #2559a7 100%);
  border-radius: 0 0 5px 5px;
  transition: 1s;
}
.gift-box.boxDown{
  animation: box-down 1s; 
  opacity:0; 
}
@keyframes box-down{
  0%{
    top:0px;
    opacity:1;
  }
  100%{
    top:50px;
    opacity:0;
  }  
}