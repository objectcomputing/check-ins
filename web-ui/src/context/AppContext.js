import React, { useEffect, useReducer, useMemo } from "react";
import { createSelector } from 'reselect';
import { getCurrentUser, updateMember, getAllMembers } from "../api/member";
import { getCheckinByMemberId, createCheckin } from "../api/checkins";
import { BASE_API_URL } from "../api/api";
import axios from "axios";

export const MY_PROFILE_UPDATE = "@@check-ins/update_profile";
export const UPDATE_USER_BIO = "@@check-ins/update_bio";
export const UPDATE_CHECKINS = "@@check-ins/update_checkins";
export const UPDATE_INDEX = "@@check-ins/update_index";
export const UPDATE_TOAST = "@@check-ins/update_toast";
export const UPDATE_CURRENT_CHECKIN = "@@check-ins/update_current_checkin";
export const UPDATE_TEAMS = "@@check-ins/update_teams";
export const UPDATE_MEMBER_PROFILES = "@@check-ins/update_member_profiles";
export const UPDATE_TEAM_MEMBERS = "@@check-ins/update_team_members";
export const UPDATE_SELECTED_PROFILE = "@@check-ins/update_selected_profile";
const SET_CSRF = "@@check-ins/update_csrf";

const AppContext = React.createContext();

const reducer = (state, action) => {
  switch (action.type) {
    case MY_PROFILE_UPDATE:
      state.userProfile = action.payload;
      break;
    case UPDATE_USER_BIO:
      state.userProfile.memberProfile.bioText = action.payload;
      updateMember(state.userProfile.memberProfile);
      break;
    case UPDATE_CHECKINS:
      state.checkins = action.payload;
      //sort by date
      state.checkins.sort(function (a, b) {
        return new Date(...a.checkInDate) - new Date(...b.checkInDate);
      });
      state.currentCheckin = state.checkins[state.checkins.length - 1];
      break;
    case SET_CSRF:
      state.csrf = action.payload;
      break;
    case UPDATE_TOAST:
      state.toast = action.payload;
      break;
    case UPDATE_TEAMS:
      state.teams = action.payload;
      break;
    case UPDATE_MEMBER_PROFILES:
      state.memberProfiles = action.payload;
      break;
    case UPDATE_TEAM_MEMBERS:
      state.teamMembers = action.payload;
      break;
    case UPDATE_CURRENT_CHECKIN:
      state.currentCheckin = action.payload;
      break;
    case UPDATE_SELECTED_PROFILE:
      const { payload } = action;
      if (state.selectedProfile !== payload) {
        state.selectedProfile = payload;
        state.currentCheckin = payload ? payload.checkIn : {};
      }
      if (payload === undefined) {
        state.checkins = [];
      }
      break;
    default:
  }
  return { ...state };
};

const initialState = {
  checkins: [],
  csrf: undefined,
  currentCheckin: {},
  teams: [],
  memberProfiles: [],
  index: 0,
  selectedProfile: undefined,
  toast: {
    severity: "",
    toast: "",
  },
  userProfile: undefined,
};

const getCheckins = async (id, pdlId, date, dispatch, csrf) => {
  const res = await getCheckinByMemberId(id, csrf);
  let data =
    res.payload && res.payload.data && res.payload.status === 200 && !res.error
      ? res.payload.data
      : null;
  if (data && data.length > 0) {
    const allComplete = data.every((checkin) => checkin.completed === true);
    if (allComplete) {
      const prevCheckinDate = data[data.length - 1].checkInDate;
      if (pdlId) {
        const res = await createCheckin({
          teamMemberId: id,
          pdlId: pdlId,
          checkInDate: date(3, prevCheckinDate),
          completed: false,
        });
        const checkin =
          res.payload && res.payload.data && !res.error
            ? res.payload.data
            : null;
        data.push(checkin);
      }
    }
  } else if (data && data.length === 0) {
    if (pdlId) {
      const res = await createCheckin({
        teamMemberId: id,
        pdlId: pdlId,
        checkInDate: date(1),
        completed: false,
      });
      const checkin =
        res.payload && res.payload.data && !res.error ? res.payload.data : null;
      data = [checkin];
    }
  }
  //without this check you get infinite checkin calls
  if (data.length > 0) {
    dispatch({ type: UPDATE_CHECKINS, payload: data });
  }
};

const AppContextProvider = (props) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const memberProfile =
    state && state.userProfile && state.userProfile.memberProfile
      ? state.userProfile.memberProfile
      : undefined;
  const id = memberProfile ? memberProfile.id : undefined;
  const selectedProfile = state && state.selectedProfile;
  const selectedId = selectedProfile ? selectedProfile.id : undefined;


  const { csrf } = state;

  useEffect(() => {
    const getCsrf = async () => {
      if (!csrf) {
        const res = await axios({
          url: `${BASE_API_URL}/csrf/cookie`,
          responseType: "text",
          withCredentials: true,
        });
        if (res && res.data) {
          dispatch({ type: SET_CSRF, payload: res.data });
        }
      }
    };
    getCsrf();
  }, [csrf]);

  useEffect(() => {
    const updateUserProfile = async () => {
      let res = await getCurrentUser(csrf);
      let profile =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profile) {
        dispatch({ type: MY_PROFILE_UPDATE, payload: profile });
      }
    };
    if (csrf) {
      updateUserProfile();
    }
  }, [csrf]);

  const date = (months, prevCheckinDate) => {
    let currentMonth = new Date().getMonth() + 1;
    let newDate = prevCheckinDate ? new Date(...prevCheckinDate) : new Date();
    if (prevCheckinDate) {
      let prevCheckinMonth = prevCheckinDate.getMonth() + 1;
      newDate.setMonth(
        newDate.getMonth() + currentMonth - prevCheckinMonth >= 3 ? 1 : months
      );
    } else {
      newDate.setMonth(newDate.getMonth() + months);
    }
    const year = newDate.getFullYear();
    const month = newDate.getMonth() + 1;
    const day = newDate.getDate();
    const hours = newDate.getHours();
    const minutes = newDate.getMinutes();
    const dateTimeArray = [year, month, day, hours, minutes, 0];
    return dateTimeArray;
  };

  useEffect(() => {
    async function getMemberProfiles() {
      let res = await getAllMembers(csrf);
      let profiles =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profiles) {
        dispatch({ type: UPDATE_MEMBER_PROFILES, payload: profiles });
      }
    }
    if (csrf) {
      getMemberProfiles();
    }
  }, [csrf]);

  useEffect(() => {
    if (selectedId && csrf) {
      getCheckins(selectedId, id, date, dispatch, csrf);
    }
  }, [csrf, selectedId, id]);

  const value = useMemo(() => {
    return { state, dispatch };
  }, [state]);
  return (
    <AppContext.Provider value={props.value || value}>
      {props.children}
    </AppContext.Provider>
  );
};

export const selectMemberProfiles = state => state.memberProfiles;
export const selectTeamMembers = state => state.teamMembers;

export const selectProfileMap = createSelector(
  selectMemberProfiles,
  (memberProfiles) => {
    if (memberProfiles && memberProfiles.length) {
      memberProfiles = memberProfiles.reduce((mappedById, profile) => {
        mappedById[profile.id] = profile;
        return mappedById;
      }, {});
    }
    return memberProfiles;
  }
);

export const selectMembersByTeamId = id => createSelector(
  selectTeamMembers,
  (teamMembers) => {
    let members = [];
    if (teamMembers && teamMembers.length) {
      members = teamMembers.filter((member) => member.teamid === id);
    }
    return members;
  }
);

export const selectMemberProfilesByTeamId = id => createSelector(
  selectMembersByTeamId(id),
  selectProfileMap,
  (members, profileMap) => members.map((member) => {
      return { ...profileMap[member.memberid], ...member };
    })
);

export { AppContext, AppContextProvider };
