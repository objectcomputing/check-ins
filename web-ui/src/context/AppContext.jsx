import React, { useEffect, useReducer, useMemo } from 'react';
import { reducer, initialState } from './reducer';
import { getCheckins, getAllCheckinsForAdmin } from './thunks';
import {
  MY_PROFILE_UPDATE,
  SET_CSRF,
  SET_ROLES,
  SET_USER_ROLES,
  UPDATE_GUILDS,
  UPDATE_MEMBER_SKILLS,
  UPDATE_MEMBER_PROFILES,
  UPDATE_TERMINATED_MEMBERS,
  UPDATE_SKILLS,
  UPDATE_CERTIFICATIONS,
  UPDATE_TEAMS,
  UPDATE_PEOPLE_LOADING,
  UPDATE_TEAMS_LOADING
} from './actions';
import {
  getCurrentUser,
  getAllMembers,
  getAllTerminatedMembers
} from '../api/member';
import {
  selectCanViewCheckinsPermission,
  selectCanViewTerminatedMembers,
} from './selectors';
import { getAllRoles, getAllUserRoles } from '../api/roles';
import { getMemberSkills } from '../api/memberskill';
import { BASE_API_URL } from '../api/api';
import { getAllGuilds } from '../api/guild';
import { getSkills } from '../api/skill';
import { getAllTeams } from '../api/team';
import {getCertifications} from "../api/certification.js";

const AppContext = React.createContext();

function getSessionCookieValue(name) {
  const cookies = document?.cookie?.split(';');
  for (let i = 0; i < cookies.length; i++) {
    const cookie = cookies[i].trim();
    if (cookie.startsWith(name + '=')) {
      return decodeURIComponent(cookie.substring(name.length + 1));
    }
  }
  return null;
}

const AppContextProvider = props => {
  const [state, dispatch] = useReducer(
    reducer,
    props?.value?.state || initialState
  );
  const userProfile =
    state && state.userProfile ? state.userProfile : undefined;
  const memberProfile =
    userProfile && userProfile.memberProfile
      ? userProfile.memberProfile
      : undefined;

  const id = memberProfile ? memberProfile.id : undefined;
  const pdlId = memberProfile ? memberProfile.pdlId : undefined;
  const {
    csrf,
    guilds,
    teams,
    memberSkills,
    memberProfiles,
    checkins,
    skills,
    certifications,
    roles,
    userRoles
  } = state;
  const url = `${BASE_API_URL}/csrf/cookie`;
  useEffect(() => {
    const getCsrf = async () => {
      if (!csrf) {
        const payload = getSessionCookieValue('_csrf');
        if (payload) {
          dispatch({ type: SET_CSRF, payload });
        } else {
          const res = await fetch(url, {
            responseType: 'text',
            credentials: 'include'
          });
          if (res && res.ok) {
            dispatch({ type: SET_CSRF, payload: await res.text() });
          }
        }
      }
    };
    getCsrf();
  }, [csrf]);

  useEffect(() => {
    async function getGuilds() {
      let res = await getAllGuilds(csrf);
      let data =
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data) {
        dispatch({ type: UPDATE_GUILDS, payload: data });
      }
    }
    if (csrf && !guilds) {
      getGuilds();
    }
  }, [csrf, guilds]);

  useEffect(() => {
    async function getTeams() {
      let res = await getAllTeams(csrf);
      let data =
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data) {
        dispatch({ type: UPDATE_TEAMS, payload: data });
        dispatch({ type: UPDATE_TEAMS_LOADING });
      }
    }
    if (csrf && !teams) {
      dispatch({ type: UPDATE_TEAMS_LOADING });
      getTeams();
    }
  }, [csrf, teams, dispatch]);

  useEffect(() => {
    const updateUserProfile = async () => {
      let res = await getCurrentUser(csrf);
      let profile =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profile) {
        dispatch({ type: MY_PROFILE_UPDATE, payload: profile });
      }
    };
    if (csrf && !userProfile) {
      updateUserProfile();
    }
  }, [csrf, userProfile]);

  useEffect(() => {
    const getAllMemberSkills = async () => {
      const res = await getMemberSkills(csrf);
      const memberSkills =
        res && res.payload && res.payload.data ? res.payload.data : null;
      if (memberSkills) {
        dispatch({ type: UPDATE_MEMBER_SKILLS, payload: memberSkills });
      }
    };
    if (csrf && !memberSkills) {
      getAllMemberSkills();
    }
  }, [csrf, memberSkills]);

  useEffect(() => {
    async function getMemberProfiles() {
      let res = await getAllMembers(csrf);
      let profiles =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profiles) {
        dispatch({ type: UPDATE_MEMBER_PROFILES, payload: profiles });
        dispatch({ type: UPDATE_PEOPLE_LOADING, payload: false });
      }
    }
    async function getTerminatedMembers() {
      let res = await getAllTerminatedMembers(csrf);
      let profiles =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profiles) {
        dispatch({ type: UPDATE_TERMINATED_MEMBERS, payload: profiles });
      }
    }
    if (csrf && userProfile && !memberProfiles) {
      dispatch({ type: UPDATE_PEOPLE_LOADING, payload: true });
      getMemberProfiles();
      if (selectCanViewTerminatedMembers(state)) {
        getTerminatedMembers();
      }
    }
  }, [csrf, userProfile, memberProfiles]);

  useEffect(() => {
    function getAllTheCheckins() {
      if (
        userProfile &&
        userProfile.permissions?.some(p =>
          p?.permission?.includes('CAN_VIEW_CHECKINS_REPORT')
        ) &&
        id &&
        csrf
      ) {
        getAllCheckinsForAdmin(dispatch, csrf);
      } else if (id && csrf && selectCanViewCheckinsPermission(state)) {
        getCheckins(id, pdlId, dispatch, csrf);
      }
    }
    if (csrf && !checkins) {
      getAllTheCheckins();
    }
  }, [csrf, pdlId, id, userProfile, checkins]);

  useEffect(() => {
    const getAllSkills = async () => {
      const res = await getSkills(csrf);
      const data =
        res &&
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data && data.length > 0) {
        dispatch({ type: UPDATE_SKILLS, payload: data });
      }
    };
    if (csrf && !skills) {
      getAllSkills();
    }
  }, [csrf, skills]);

  useEffect(() => {
    const getAllCertifications = async () => {
      const res = await getCertifications(csrf);
      const data =
          res &&
          res.payload &&
          res.payload.data &&
          res.payload.status === 200 &&
          !res.error
              ? res.payload.data
              : null;
      if (data && data.length > 0) {
        dispatch({ type: UPDATE_CERTIFICATIONS, payload: data });
      }
    };
    if (csrf && !certifications) {
      getAllCertifications();
    }
  }, [csrf, certifications]);

  useEffect(() => {
    const getRoles = async () => {
      const res = await getAllRoles(csrf);
      const data =
        res &&
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data && Array.isArray(data) && data.length > 0) {
        dispatch({ type: SET_ROLES, payload: data });
      }
    };
    if (csrf && !roles) {
      getRoles();
    }
  }, [csrf, roles]);

  useEffect(() => {
    const getUserRoles = async () => {
      // make call to the API
      let res = await getAllUserRoles(csrf);
      return res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
        ? res.payload.data
        : null;
    };

    if (csrf && !userRoles) {
      getUserRoles().then(userRoles => {
        dispatch({ type: SET_USER_ROLES, payload: userRoles });
      });
    }
  }, [csrf, userRoles]);

  const value = useMemo(() => {
    return { state, dispatch };
  }, [state]);
  return (
    <AppContext.Provider value={props.value || value}>
      {props.children}
    </AppContext.Provider>
  );
};

export { AppContext, AppContextProvider };
