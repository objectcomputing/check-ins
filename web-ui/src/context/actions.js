export const MY_PROFILE_UPDATE = "@@check-ins/update_profile";
export const UPDATE_USER_BIO = "@@check-ins/update_bio";
export const UPDATE_CHECKINS = "@@check-ins/update_checkins";
export const UPDATE_CHECKIN = "@@check-ins/update_checkin";
export const ADD_CHECKIN = "@@check-ins/add_checkin";
export const UPDATE_TOAST = "@@check-ins/update_toast";
export const UPDATE_MEMBER_SKILLS = "@@check-ins/update_member_skills";
export const DELETE_MEMBER_SKILL = "@@check-ins/delete_member_skill";
export const ADD_MEMBER_SKILL = "@@check-ins/add_member_skill";
export const UPDATE_TEAMS = "@@check-ins/update_teams";
export const UPDATE_MEMBER_PROFILES = "@@check-ins/update_member_profiles";
export const UPDATE_TEAM_MEMBERS = "@@check-ins/update_team_members";
export const ADD_SKILL = "@@check-ins/add-skill";
export const DELETE_SKILL = "@@check-ins/delete-skill";
export const UPDATE_SKILL = "@@check-ins/update_skill";
export const UPDATE_SKILLS = "@@check-ins/update_skills";
export const ADD_TEAM = "@@check-ins/add_team";
export const SET_CSRF = "@@check-ins/update_csrf";
export const SET_ROLES = "@@check-ins/set_roles";
