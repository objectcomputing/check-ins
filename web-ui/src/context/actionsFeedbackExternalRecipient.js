export const SET_CSRF = '@@check-ins/update_csrf';
export const UPDATE_TOAST = '@@check-ins/update_toast';

