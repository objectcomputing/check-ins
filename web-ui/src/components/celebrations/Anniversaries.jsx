import React, { useContext } from "react";
import { Link } from "react-router-dom";
import { AppContext } from "../../context/AppContext";

import { styled } from "@mui/material/styles";
import { getAvatarURL } from "../../api/api.js";
import {
  selectMemberProfilesLoading,
  selectProfile,
} from "../../context/selectors";
import SkeletonLoader from "../skeleton_loader/SkeletonLoader";

import { Card, CardHeader } from "@mui/material";
import Avatar from "@mui/material/Avatar";

import { TextField, Grid } from "@mui/material";

import "./Anniversaries.css";

import { Typography } from "@mui/material";

const PREFIX = "MemberSummaryCard";
const classes = {
  header: `${PREFIX}-header`,
};
const Root = styled("div")({
  [`& .${classes.search}`]: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
  },
  [`& .${classes.searchInput}`]: {
    width: "20em",
  },
  [`& .${classes.members}`]: {
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "space-evenly",
    width: "100%",
  },
});

const Anniversaries = ({ anniversaries }) => {
  const { state } = useContext(AppContext);

  const loading = selectMemberProfilesLoading(state);

  const createAnniversaryCards = anniversaries.map((anniv, index) => {
    let user = selectProfile(state, anniv.userId);
    return (
      <Card className={"anniversaries-card"} key={index}>
        <Link
          style={{ color: "black", textDecoration: "none" }}
          to={`/profile/${anniv.userId}`}
        >
          <CardHeader
            className={classes.header}
            title={
              <Typography variant="h5" component="h2">
                {anniv.name} ({anniv.anniversary})
              </Typography>
            }
            subheader={
              <Typography color="textSecondary" component="h3">
                Thank you for{" "}
                <span>
                  {anniv.yearsOfService.toFixed(0)}
                  {anniv.yearsOfService > 1 ? " years" : " year"}{" "}
                </span>
                of service!
              </Typography>
            }
            disableTypography
            avatar={
              <Avatar
                className={"celebrations-avatar"}
                src={getAvatarURL(user?.workEmail)}
              />
            }
          />
        </Link>
      </Card>
    );
  });

  return (
    <div className="anniversaries">
      <div className={"anniversary-gift box"}>
        <div className={"anniversary-gift-top open"}></div>
        <div className={"anniversary-gift-text open"}></div>
        <div className={"anniversary-gift-box"}></div>
      </div>
      <Root>
        <div className="anniversary-title">
          <h2>Anniversaries</h2>
        </div>
        <Grid container columns={6} spacing={3}>
          <Grid item className={classes.members}>
            {loading
              ? Array.from({ length: 20 }).map((_, index) => (
                  <SkeletonLoader key={index} type="people" />
                ))
              : !loading
              ? createAnniversaryCards
              : null}
          </Grid>
        </Grid>
      </Root>
    </div>
  );
};

export default Anniversaries;
