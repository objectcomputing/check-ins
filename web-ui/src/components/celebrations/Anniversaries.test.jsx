import React from 'react';
import Anniversaries from './Anniversaries';
import { AppContextProvider } from '../../context/AppContext';
import { BrowserRouter } from 'react-router-dom';

const anniversaries = [
  {
    name: 'Jesse Hanner',
    anniversary: '12/20/2012',
    yearsOfService: 10.03,
    userId: '67dc3a3b-5bfa-4759-997a-fb6bac98dcf3'
  },
  {
    name: 'Mohit Bhatia',
    anniversary: '12/10/2016',
    yearsOfService: 6.06,
    userId: 'b2d35288-7f1e-4549-aa2b-68396b162490'
  },
  {
    name: 'Zack Brown',
    anniversary: '12/29/2019',
    yearsOfService: 3.01,
    userId: '43ee8e79-b33d-44cd-b23c-e183894ebfef'
  }
];

it('renders correctly', () => {
  snapshot(
    <AppContextProvider>
      <BrowserRouter>
        <Anniversaries anniversaries={anniversaries} />
      </BrowserRouter>
    </AppContextProvider>
  );
});
