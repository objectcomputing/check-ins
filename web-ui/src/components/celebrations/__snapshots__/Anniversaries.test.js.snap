// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.emotion-0 .undefined {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-evenly;
  -ms-flex-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  justify-content: space-evenly;
  width: 100%;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -24px;
  width: calc(100% + 24px);
  margin-left: -24px;
}

.emotion-1>.MuiGrid-item {
  padding-top: 24px;
}

.emotion-1>.MuiGrid-item {
  padding-left: 24px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-3 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 16px;
}

.emotion-6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-7 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
  color: transparent;
  text-indent: 10000px;
}

.emotion-8 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
}

.emotion-10 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.6);
}

<div
  className="anniversaries"
>
  <div
    className="anniversary-gift box"
  >
    <div
      className="anniversary-gift-top open"
    />
    <div
      className="anniversary-gift-text open"
    />
    <div
      className="anniversary-gift-box"
    />
  </div>
  <div
    className="emotion-0"
  >
    <div
      className="anniversary-title"
    >
      <h1>
        Anniversaries
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 emotion-1"
    >
      <div
        className="MuiGrid-root MuiGrid-item emotion-2"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root anniversaries-card emotion-3"
        >
          <a
            href="/profile/67dc3a3b-5bfa-4759-997a-fb6bac98dcf3"
            onClick={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
          >
            <div
              className="MuiCardHeader-root MemberSummaryCard-header emotion-4"
            >
              <div
                className="MuiCardHeader-avatar emotion-5"
              >
                <div
                  className="MuiAvatar-root MuiAvatar-circular celebrations-avatar emotion-6"
                >
                  <img
                    className="MuiAvatar-img emotion-7"
                    src="http://localhost:8080/services/member-profiles/member-photos/undefined"
                  />
                </div>
              </div>
              <div
                className="MuiCardHeader-content emotion-8"
              >
                <h2
                  className="MuiTypography-root MuiTypography-h5 emotion-9"
                >
                  Jesse Hanner
                   (
                  12/20/2012
                  )
                </h2>
                <h3
                  className="MuiTypography-root MuiTypography-body1 emotion-10"
                >
                  Thank you for
                   
                  <span>
                    10
                     years
                     
                  </span>
                  of service!
                </h3>
              </div>
            </div>
          </a>
        </div>
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root anniversaries-card emotion-3"
        >
          <a
            href="/profile/b2d35288-7f1e-4549-aa2b-68396b162490"
            onClick={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
          >
            <div
              className="MuiCardHeader-root MemberSummaryCard-header emotion-4"
            >
              <div
                className="MuiCardHeader-avatar emotion-5"
              >
                <div
                  className="MuiAvatar-root MuiAvatar-circular celebrations-avatar emotion-6"
                >
                  <img
                    className="MuiAvatar-img emotion-7"
                    src="http://localhost:8080/services/member-profiles/member-photos/undefined"
                  />
                </div>
              </div>
              <div
                className="MuiCardHeader-content emotion-8"
              >
                <h2
                  className="MuiTypography-root MuiTypography-h5 emotion-9"
                >
                  Mohit Bhatia
                   (
                  12/10/2016
                  )
                </h2>
                <h3
                  className="MuiTypography-root MuiTypography-body1 emotion-10"
                >
                  Thank you for
                   
                  <span>
                    6
                     years
                     
                  </span>
                  of service!
                </h3>
              </div>
            </div>
          </a>
        </div>
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root anniversaries-card emotion-3"
        >
          <a
            href="/profile/43ee8e79-b33d-44cd-b23c-e183894ebfef"
            onClick={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
          >
            <div
              className="MuiCardHeader-root MemberSummaryCard-header emotion-4"
            >
              <div
                className="MuiCardHeader-avatar emotion-5"
              >
                <div
                  className="MuiAvatar-root MuiAvatar-circular celebrations-avatar emotion-6"
                >
                  <img
                    className="MuiAvatar-img emotion-7"
                    src="http://localhost:8080/services/member-profiles/member-photos/undefined"
                  />
                </div>
              </div>
              <div
                className="MuiCardHeader-content emotion-8"
              >
                <h2
                  className="MuiTypography-root MuiTypography-h5 emotion-9"
                >
                  Zack Brown
                   (
                  12/29/2019
                  )
                </h2>
                <h3
                  className="MuiTypography-root MuiTypography-body1 emotion-10"
                >
                  Thank you for
                   
                  <span>
                    3
                     years
                     
                  </span>
                  of service!
                </h3>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
